iterator.go

```diff
@@ -25,14 +25,29 @@ import (
 //	         return err
 //	     }
 type Iterator struct {
-	c *C.rocksdb_iterator_t
+	c              *C.rocksdb_iterator_t
+	iterUpperBound []byte
+	iterLowerBound []byte
+	timestamp      []byte
+	timestampStart []byte
 }
 
 // NewNativeIterator creates a Iterator object.
 func newNativeIterator(c *C.rocksdb_iterator_t) *Iterator {
 	return &Iterator{c: c}
 }
 
+// newNativeIteratorWithReadOptions creates a Iterator object with ReadOptions.
+func newNativeIteratorWithReadOptions(c *C.rocksdb_iterator_t, opts *ReadOptions) *Iterator {
+	return &Iterator{
+		c:              c,
+		iterUpperBound: opts.iterUpperBound,
```

Comment:
Please clone the byte slice as well

Reply:
It's by purpose to hold the reference of byte slice in iterator. So to ensure they're not released early when using iterator.

Reply:
@mmsqe I mean you must `clone` the slice, not just keep the reference. In this way, modifying opts.iterUpperBound (outside) will not harm the iterator!

```go
iterUpperBound: clone(opts.iterUpperBound)


func clone(b []byte) []byte {
   cloned := make([]byte, len(b))
   copy(cloned, b)
   return cloned
}
```

Reply:
It'll lost the reference, you might try `TestIteratorCFWithTSWhenNoDestroy`

Reply:
@mmsqe 

of course, you must use cloned slice as reference. In another word, please clone `opts`, use cloned one as native iterator's params.